codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java

```diff
@@ -306,6 +308,7 @@ public InterfaceHttpData getBodyHttpData(String name) {
         return null;
     }
 
+    public static int TEST_TEMP_ITEM = 1;
```

Comment:
Use final here, it would change a lot how the JIT would optimize it.
Use a sys property to set this and just use different runs with different JVMs (probably using a good profiler + JMH bench would be ideal to be use of the impact/meaning of changes)

Reply:
@franz1981 Thank you !
I agree that making it final shall be done to achieve final checks. This variable was only intend to allow 3 kinds of tests, not to be keeped of course in final code (and if final, I couldn't change the value of course).

Now, using a final here, 3 different runs (changing value of course) and a good profiler could be far better and ideal, I agree. 

However, as I said, when using a profiler (the one I have is from Oracle - VisualVM-, and I believe it is not the best around, but the only one I've got), the performances drop severely (even in DISCARDED mode), therefore nothing to look at correctly. For my other tasks, this one is great, giving the information I need and with not that much impact. But there, the performances are almost all at the same level than PARANOID, even in DISCARDED or SIMPLE mode.

Can you suggest some tools (free, sadly) that could help in this research?

Reply:
Sure!

https://github.com/jvm-profiling-tools/async-profiler is probably one of the most complete, accurate and cheap (;)) ones I know.
Highly suggested to add -XX:+UnlockDiagnosticVMOptions -XX:+DebugNonSafepoints

Reply:
@njhill now you read my mind bud?:)